examples/platforms/simulation/logging.c

```diff
@@ -44,21 +45,81 @@
 #include "utils/code_utils.h"
 
 #if (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED)
-OT_TOOL_WEAK void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
+
+static FILE *sLogFile = NULL;
+
+void platformLoggingSetFileName(const char *aName)
+{
+    if (sLogFile != NULL)
+    {
+        fclose(sLogFile);
+    }
+
+    sLogFile = fopen(aName, "wt");
+
+    if (sLogFile == NULL)
+    {
+        fprintf(stderr, "Failed to open log file '%s', errno %d\r\n", aName, errno);
+        exit(EXIT_FAILURE);
+    }
+}
+
+void platformLoggingInit(const char *aName)
+{
+    if (sLogFile == NULL)
+    {
+        openlog(aName, LOG_PID, LOG_USER);
+        setlogmask(setlogmask(0) & LOG_UPTO(LOG_NOTICE));
+    }
+    else
+    {
+        fprintf(sLogFile, "OpenThread logs\r\n");
+        fprintf(sLogFile, "- Program:  %s\r\n", aName);
+        fprintf(sLogFile, "- Platform: simulation\r\n");
+        fprintf(sLogFile, "- Node ID:  %lu\r\n", (unsigned long)gNodeId);
+        fprintf(sLogFile, "\r\n");
+    }
+}
+
+void platformLoggingDeinit(void)
+{
+    if (sLogFile != NULL)
+    {
+        fclose(sLogFile);
```

Comment:
should we set `sLogFile = NULL` as well?

Reply:
Good point. Updated in new push.